.github/workflows/ci.yml

```diff
@@ -337,8 +337,8 @@ jobs:
           submodules: true
       - name: Run tests
         run: |
-          python3 -m pip install --upgrade --requirement requirements-dev.txt
-          python3 -m pip install . --verbose
+          python3 -m pip install --upgrade pip
```

Comment:
semi related: do we still think upgrading pip is a good idea. i remember we had some issues in the past year with python blocking global pip upgrade in some cases and instead pointing to upgrading pip within venv

Reply:
I'll try taking these out and see what happens ... and it failed.

tried moving the job to ubuntu-24.04, which would have newer pip ... and that failed too due to formatting errors raised by sphinx.

The docs are sadly very fragile. Don't have time to diagnose. Just making it work for now :(